### PR TITLE
adding new color property in Text and Link component

### PR DESCRIPTION
**Pull request by tejasnikam**
Fix issues [#2286 ](https://github.com/jpmorganchase/salt-ds/issues/2286)

### CODE REVIEW

File: packages/core/src/text/Text.tsx [hunk 1]
```diff
@@ -37,6 +37,7 @@ export type TextProps<T extends ElementType> = PolymorphicComponentPropWithRef<
      * Change text color palette
      */
     variant?: "primary" | "secondary";
+    color?: "primary" | "secondary";
```
Comment (joshwooding) [inline packages/core/src/text/Text.tsx:40]: We should copy Icon here

https://github.com/jpmorganchase/salt-ds/blob/5e48416c048e474b3750f405d677b608853b10ef/packages/icons/src/icon/Icon.tsx#L20

**Review by joshwooding — CHANGES_REQUESTED** [review]
Thanks for this, we need to do a bit more work here:

- Add two Storybook stories one for Link and Text showcasing the prop
- Update our QA stories to prevent visual regressions

There's some other steps too but we can tackle those after

**Comment by tejasnikam** [comment]
> Thanks for this, we need to do a bit more work here:
> 
> * Add two Storybook stories one for Link and Text showcasing the prop
> * Update our QA stories to prevent visual regressions
> 
> There's some other steps too but we can tackle those after

Thanks for the quick review

-  Have added two Storybook stories one for Link and Text showcasing the prop
-  Updated QA stories to prevent visual regressions
-  Also updated documentation

File: packages/core/src/text/Text.tsx (hunk 1 above)
Comment (tejasnikam) [reply packages/core/src/text/Text.tsx:40]: Needful has been done

**Comment by tejasnikam** [comment]
Hello Team,
Any updates on this.

**Review by joshwooding — APPROVED** [review]

**Review by Fercas123 — APPROVED** [review]

**Pull request merged**